### PR TITLE
Milestone 8. BlogAppRoR Project integration-specs Branch 

### PR DESCRIPTION
**Pull request by otmanTR**
In this branch following changes done:

-Before n+1 problem fixed
Before
-After n+1 problem fixed
After

-Add integration spec tests to user index page:
  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.

-Add integration spec tests to user show page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.

- Add integration spec tests to users post index page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the first comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.

- Add integration spec tests to post show page:
  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each commentor.
  - I can see the comment each commentor left.

My coding partner didn't show up and the student support team unpair us. I am alone now
unpair

### CODE REVIEW

File: spec/integration/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    it 'shows the number of likes on the post' do
+      expect(page).to have_content("Likes: #{@first_post.likes_counter}")
+    end
+
+    it "when click on a post, it redirects to the post's show page" do
+      visit user_posts_path(@first_user)
+      click_link @first_post.title
+      expect(current_path).to match(user_posts_path(@first_user.id))
+    end
+  end
```
Comment (thecodechaser) [inline spec/integration/posts_spec.rb]: - [x] Please make sure to include the below test for the posts index page as requested in the project requirements.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @otmanTR👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- All tests are passed. ✔️
- Gitflow is followed. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️

- [ ] Please make sure to solve the N+1 problem when fetching all posts and their comments for a user as requested in the project requirements. Example in the posts#index actions `@posts = @user.posts.includes(:comments)` instead of `@posts = Post.all`.
- [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/integration/posts_spec.rb (hunk 1 above)
Comment (otmanTR) [reply spec/integration/posts_spec.rb]: Thank you!

**Review by gemmen29 — APPROVED** [review]
Hi @otmanTR, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**